# Predict the Prices of Stocks
Weekly challenge: Week 13  
Date: 10/17/2022

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

In [2]:
import os

In [3]:
# List of available stock data files
os.listdir(r'C:\Users\unpat\_Projects\Challenge_AprojectAweek_data_files\Week_13\individual_stocks_5yr')

['.DS_Store',
 'AAL_data.csv',
 'AAPL_data.csv',
 'AAP_data.csv',
 'ABBV_data.csv',
 'ABC_data.csv',
 'ABT_data.csv',
 'ACN_data.csv',
 'ADBE_data.csv',
 'ADI_data.csv',
 'ADM_data.csv',
 'ADP_data.csv',
 'ADSK_data.csv',
 'ADS_data.csv',
 'AEE_data.csv',
 'AEP_data.csv',
 'AES_data.csv',
 'AET_data.csv',
 'AFL_data.csv',
 'AGN_data.csv',
 'AIG_data.csv',
 'AIV_data.csv',
 'AIZ_data.csv',
 'AJG_data.csv',
 'AKAM_data.csv',
 'ALB_data.csv',
 'ALGN_data.csv',
 'ALK_data.csv',
 'ALLE_data.csv',
 'ALL_data.csv',
 'ALXN_data.csv',
 'AMAT_data.csv',
 'AMD_data.csv',
 'AME_data.csv',
 'AMGN_data.csv',
 'AMG_data.csv',
 'AMP_data.csv',
 'AMT_data.csv',
 'AMZN_data.csv',
 'ANDV_data.csv',
 'ANSS_data.csv',
 'ANTM_data.csv',
 'AON_data.csv',
 'AOS_data.csv',
 'APA_data.csv',
 'APC_data.csv',
 'APD_data.csv',
 'APH_data.csv',
 'APTV_data.csv',
 'ARE_data.csv',
 'ARNC_data.csv',
 'ATVI_data.csv',
 'AVB_data.csv',
 'AVGO_data.csv',
 'AVY_data.csv',
 'AWK_data.csv',
 'AXP_data.csv',
 'AYI_data.csv',

In [4]:
len(os.listdir(r'C:\Users\unpat\_Projects\Challenge_AprojectAweek_data_files\Week_13\individual_stocks_5yr'))

506

-- We have stock data for 506 - 1 = 505 companies  
-- We will analyze stocks of the following companies  
-- Apple, Amazon, IBM, Google

In [5]:
company_list = ['AAPL_data.csv','AMZN_data.csv','IBM_data.csv','GOOG_data.csv']

In [6]:
# Creating a blank dataframe to store the stock data of the above companies
all_data = pd.DataFrame()

In [7]:
# Directory path of the data files
path = r'C:\Users\unpat\_Projects\Challenge_AprojectAweek_data_files\Week_13\individual_stocks_5yr'

In [9]:
# Reading the data files
for file in company_list:
    current_df = pd.read_csv(filepath_or_buffer = path + '/' + file)
    
    # storing (concatenating) the data from each file into the blank dataframe
    all_data = pd.concat([all_data, current_df])

In [10]:
# Shape of the final dataframe 'all_data'
all_data.shape

(4752, 7)

In [11]:
all_data.head(3)

date     open     high      low    close     volume  Name
0  2013-02-08  67.7142  68.4014  66.8928  67.8542  158168416  AAPL
1  2013-02-11  68.0714  69.2771  67.6071  68.5614  129029425  AAPL
2  2013-02-12  68.5014  68.9114  66.8205  66.8428  151829363  AAPL